portal/account/index_reset.php

```diff
@@ -0,0 +1,170 @@
+<?php
+/**
+ * Credential Changes
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author    Jerry Padgett <sjpadgett@gmail.com>
+ * @copyright Copyright (c) 2019 Jerry Padgett <sjpadgett@gmail.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+$ignoreAuth_onsite_portal_two = $ignoreAuth = 0;
+// Will start the (patient) portal OpenEMR session/cookie.
+require_once(dirname(__FILE__) . "/../../src/Common/Session/SessionUtil.php");
+OpenEMR\Common\Session\SessionUtil::portalSessionStart();
+
+$landingpage = "./../index.php?site=" . urlencode($_SESSION['site_id']);
+// kick out if patient not authenticated
+if (isset($_SESSION['pid']) && isset($_SESSION['patient_portal_onsite_two'])) {
+    $ignoreAuth_onsite_portal_two = true;
+} else {
+    OpenEMR\Common\Session\SessionUtil::portalSessionCookieDestroy();
+    header('Location: ' . $landingpage . '&w');
+    exit;
+}
+require_once(dirname(__FILE__) . '/../../interface/globals.php');
+require_once("$srcdir/authentication/common_operations.php");
+require_once(dirname(__FILE__) . "/../lib/appsql.class.php");
+
+use OpenEMR\Core\Header;
+
+$logit = new ApplicationTable();
+//exit if portal is turned off
+if (!(isset($GLOBALS['portal_onsite_two_enable'])) || !($GLOBALS['portal_onsite_two_enable'])) {
+    echo xlt('Patient Portal is turned off');
+    exit;
+}
+
+$_SESSION['credentials_update'] = 1;
+DEFINE("TBL_PAT_ACC_ON", "patient_access_onsite");
+DEFINE("COL_PID", "pid");
+DEFINE("COL_POR_PWD", "portal_pwd");
+DEFINE("COL_POR_USER", "portal_username");
+DEFINE("COL_POR_LOGINUSER", "portal_login_username");
+DEFINE("COL_POR_SALT", "portal_salt");
+DEFINE("COL_POR_PWD_STAT", "portal_pwd_status");
+
+$sql = "SELECT " . implode(",", array(COL_ID, COL_PID, COL_POR_PWD, COL_POR_USER, COL_POR_LOGINUSER, COL_POR_SALT, COL_POR_PWD_STAT)) .
+    " FROM " . TBL_PAT_ACC_ON . " WHERE pid = ?";
+
+$auth = privQuery($sql, array($_SESSION['pid']));
+$code_current = trim($_POST['pass_current']);
+$current_encoded = oemr_password_hash($code_current, $auth[COL_POR_SALT]);
+if (isset($_POST['pass_current']) && $current_encoded != $auth[COL_POR_PWD]) {
+    $errmsg = xlt("Invalid Current Credentials Error.") . xlt("Current password.");
+    $logit->portalLog('Credential update attempt', '', ($_POST['uname'] . ':invalid password'), '', '0');
+    die($errmsg);
+}
```

Comment:
We are already in an authorized session so isset($_POST['pass_current']) will always be true, empty or not.  Unless the session is hijacked or someone is piggybacking off users current user agents local session. I'm not that paranoid.
If post is empty it still fails check.
I might be able to lock it down some more but I don't see a need.

Reply:
I live in a constant state of paranoia (ok, I am being a bit dramatic :) )
Check out my comment I just made below. I can bypass this if I submit a post request that does not include the $_POST['pass_current'] variable.